packages/react-search-ui-views/src/__tests__/view-helpers/getUrlSanitizer.test.js

```diff
@@ -0,0 +1,95 @@
+import { getUrlSanitizer } from "../../view-helpers";
+import URL from "core-js-pure/features/url";
+
+describe("getUrlSanitizer", () => {
+  let url;
+  let currentLocation;
+
+  beforeEach(() => {
+    url = "";
+    currentLocation = "";
+  });
+
+  function subject() {
+    return getUrlSanitizer(URL, currentLocation)(url);
+  }
+
+  describe("when valid url with http", () => {
+    beforeEach(() => {
+      url = "http://www.example.com/";
+      currentLocation = "http://www.mysite.com";
+    });
+
+    it("should allow it", () => {
+      expect(subject()).toEqual(url);
+    });
+  });
+
+  describe("when valid url with https", () => {
+    beforeEach(() => {
+      url = "https://www.example.com/";
+      currentLocation = "http://www.mysite.com";
+    });
+
+    it("should allow it", () => {
+      expect(subject()).toEqual(url);
+    });
+  });
+
+  describe("when relative url", () => {
+    beforeEach(() => {
+      url = "/item/1234";
+      currentLocation = "http://www.mysite.com";
+    });
+
+    it("should allow it", () => {
+      expect(subject()).toEqual("http://www.mysite.com/item/1234");
+    });
+  });
+
+  describe("when the protocol is javascript", () => {
```

Comment:
Just because I'm paranoid, can we also include a test for `" javascript://test%0aalert(document.domain)"` (with a space)? Seems like [a non-obvious case](https://medium.com/javascript-security/avoiding-xss-in-react-is-still-hard-d2b5c7ad9412) that could trip up some folks when a location isn't specified.

Reply:
https://github.com/elastic/search-ui/pull/481/commits/ac3b5bc88ce4f7bc188c332e11d11647f76b76ac